### PR TITLE
Add function to select latest version of boot image

### PR DESCRIPTION
**Pull request by aaannz**
Fixes https://github.com/SUSE/spacewalk/issues/17207

### CODE REVIEW

File: image-sync-formula/_modules/image_sync.py [hunk 1]
```diff
@@ -97,3 +97,11 @@ def deleted_boot_images(boot_images_in_use, arch_list = ['x86_64', 'i586', 'i686
...
             res.append((boot_image_id, boot_image_data))
     return res
+
+def get_default_boot_image(boot_images_in_use: object) -> str:
+    """
+    Return boot image with the highest version number
+
+    boot_images_in_use list is in format "image_name-image_version"
+    """
+    return sorted(boot_images_in_use,key=lambda s:(s.rpartition("-"))[0], reverse=True)[0]
```
[the remark this answers is not in the document]
Comment (aaannz) [reply image-sync-formula/_modules/image_sync.py:107]: Right! My mistake, should be [2]. Regexp of course would work as well. There is also issue that this is string comparison and not numeric one or semantic versioning comparison. But AFAIK users are free to use whatever versioning scheme they like (need to check this claim in kiwi docu).

I don't know what other choice we have, how we will know what image should be the default one? Please take a look at the bug report this is trying to fix. We need to ensure that when there is update for saltboot and image is rebuild with higher version updated initrd is used as default.

[the remark this answers is not in the document]
Comment (aaannz) [reply image-sync-formula/_modules/image_sync.py:107]: Checked kiwi documentation ( https://osinside.github.io/kiwi/image_description/elements.html?highlight=version#preferences-version ) and version has to be X.Y.Z with all of them being numeric value. So semantic versioning comparison is way to go.

**Comment by aaannz** [comment]
- changed to correctly use X.Y.Z(-Rev) format
- added option to disable this behavior using pillar

**Review by nadvornik — APPROVED** [review]
OK, let's try it.

**Pull request merged**